core/domain/exp_services.py

```diff
@@ -1578,53 +1650,92 @@ def regenerate_exploration_and_contributors_summaries(exploration_id):
         exploration_id: str. ID of the exploration.
     """
     exploration = exp_fetchers.get_exploration_by_id(exploration_id)
-    exp_summary = _compute_summary_of_exploration(exploration)
-    exp_summary.contributors_summary = (
-        compute_exploration_contributors_summary(exp_summary.id))
-    save_exploration_summary(exp_summary)
-
-
-def _compute_summary_of_exploration(exploration):
-    """Create an ExplorationSummary domain object for a given Exploration
-    domain object and return it.
+    exp_rights = rights_manager.get_exploration_rights(
+        exploration_id, strict=True)
+    exp_summary = exp_fetchers.get_exploration_summary_by_id(
+        exploration_id, strict=True)
+    updated_exp_summary = update_exploration_summary(
+        exploration, exp_rights, exp_summary)
+    updated_exp_summary.contributors_summary = (
+        compute_exploration_contributors_summary(updated_exp_summary.id))
+    save_exploration_summary(updated_exp_summary)
+
+
+def update_exploration_summary(
+    exploration,
+    exp_rights,
+    exp_summary,
+    skip_exploration_model_last_updated=False
+):
+    """Updates an exploration summary domain object from a given exploration
+    and its rights.
 
     Args:
         exploration: Exploration. The exploration whose summary is to be
             computed.
+        exp_rights: ActivityRights. The exploration rights model, used
+            to compute summary.
+        exp_summary: ExplorationSummary. The exploration summary
+            model whose summary needs to be recomputed.
+        skip_exploration_model_last_updated: bool. Whether the update of
+            exploration_model_last_updated should be skipped.
+            The exploration_model_last_updated is computed from the last human
+            update of the exploration. This can be skipped  in cases where we
+            know that there was no human exploration update.
 
     Returns:
         ExplorationSummary. The resulting exploration summary domain object.
     """
-    exp_rights = exp_models.ExplorationRightsModel.get_by_id(exploration.id)
-    exp_summary_model = exp_models.ExpSummaryModel.get_by_id(exploration.id)
-    if exp_summary_model:
-        old_exp_summary = exp_fetchers.get_exploration_summary_from_model(
-            exp_summary_model)
-        ratings = old_exp_summary.ratings or feconf.get_empty_ratings()
-        scaled_average_rating = get_scaled_average_rating(ratings)
+    scaled_average_rating = get_scaled_average_rating(exp_summary.ratings)
+
+    if skip_exploration_model_last_updated:
+        exploration_model_last_updated = (
+            exp_summary.exploration_model_last_updated)
     else:
-        ratings = feconf.get_empty_ratings()
-        scaled_average_rating = feconf.EMPTY_SCALED_AVERAGE_RATING
+        exploration_model_last_updated = datetime.datetime.fromtimestamp(
```

Comment:
Since this is being calculated independently, wouldn't this potentially cause a discrepancy between exp_summary.exploration_model_last_updated and exploration.last_updated, when the two values are actually supposed to be in sync?

Reply:
Hmm, probably, but I'm not sure if we expect them to be in sync or not. As this is not a newly added thing (it was calculated like this before) I propose keeping it as is for now with an TODO an revisit this later when we have validations and this will probably come up.

Reply:
OK with me, go ahead and add a TODO + issue and I'll approve. Thanks!

Reply:
Done.